### ISSUE TITLE
Make it work outside cloudflare

### ISSUE THREAD

**Issue by slorquet** [cloudflare/wildebeest#84]
Because mastodon and the fediverse are federated, not centralized on a particular provider platform.

Discord killed the concept of "server" please dont kill federation.

**Comment by mjgardner** [comment]
This is hilarious. Next you'll ask for AWS to port their [business apps](https://aws.amazon.com/business-applications/) to other clouds.

**Comment by slorquet** [comment]
Mastodon is not a business app.

**Comment by mjgardner** [comment]
> Mastodon is not a business app.

But:
* wildebeest is an ActivityPub server that also implements some of Mastodon's API. ActivityPub > Mastodon
* Cloudflare *is* a business

**Comment by slorquet** [comment]
Come on, free software that runs in a walled garden is not really free.

The real goal here seem to be to attract more customers to cloudflare products by taking advantage of a platform which was primarily established as a non commercial network of people that dislike the commercial nature of social networks,

I take bets that the instances that will be created on cloudflare will probably be free first, but will soon grow too big to stay in the free tier, so people will have to pay to keep them. Or they will display ads? This feels bad.

Time will tell.

**Comment by NoaHimesaka1873** [comment]
> Come on, free software that is made with the purpose of running in a single walled garden is not really free.
> 
> The real goal here seem to be to attract more customers to cloudflare products by taking advantage of a platform which was primarily established as a non commercial network of people that dislike the commercial nature of social networks,
> 
> I take bets that the instances that will be created on cloudflare will probably be free first, but will soon grow too big to stay in the free tier, so people will have to pay to keep them. Or they will display ads? This feels bad.
> 
> Time will tell.

It's not that we don't have other great APub implementations. Mastodon, Akkoma, and others already exist and gives great APub experience. There is lots of alternatives. If you don't want to use walled garden products, you can just ignore this. Also they don't say it's free, you have to use their images which is paid option only. I personally see this as an easy way to run small personal instances.  If you can bear costs, and have little sysadmin expertise, this seems like a great way!

**Comment by mjgardner** [comment]
> The real goal here seem to be to attract more customers to cloudflare products

Yes. Why else?

> by taking advantage of a platform which was primarily established as a non commercial network of people that dislike the commercial nature of social networks,

The only way to implement a decentralized network acceptable use policy is through the federation choices of node operators. Otherwise, you are either asking for a central policy body (e.g., the National Science Foundation and its  NSFnet backbone networks from 1985 to 1995 which prohibited commercial Internet traffic) or you are asking to impose a licensing agreement on protocol implementations, something that is anathema to the W3C.

Basically, you don't get to be open and closed at the same time.

> I take bets that the instances that will be created on cloudflare will probably be free first, but will soon grow too big to stay in the free tier, so people will have to pay to keep them.

<i lang="la">Mirabile dictu</i>, a business model is born.

> Or they will display ads?

That is one unimaginative possibility.

> This feels bad.

I missed the part where bad feelings become bug reports.

**Comment by raingloom** [comment]
The point of this issue is that if the developers got a good faith request to make their implementation work outside their walled garden.

If they reject it, there is a record of it.

My 2 cents is this could be a classic GMail scenario.  Massive corporate players are trying to become huge nodes in the federation so that they can start dictating what other implementations should do.

The only way to prevent that is a) by blocking them early on so they don't get to grow huge, or b) them making the implementation self-hostable outside their infrastructure, thereby giving up their monopoly position voluntarily.

**Comment by ocdtrekkie** [comment]
Obviously Cloudflare employees are building a Cloudflare-based implementation probably on company time which will drive some amount of Cloudflare usage. There is no sane or sensible reason to believe this project would have any reason to support otherwise.

As an open standard, it's fantastic we can have numerous implementations of ActivityPub that operate in all sorts of different ways.

As with other open source software, if you'd like to modify it to run it somewhere else, you are free to fork it, modify it to do so, and it's even possible at Cloudflare's discretion they might accept the PR upstream.

But this issue seems like an act of essentially trolling, and I think Cloudflare should probably both close and lock it.

**Comment by raingloom** [comment]
> I think we are very far from a point where a corporate implementation has standards-dictating authority.

And we should only get farther. :)

What the original devs intend matters little, *if* it gets enough steam, we have the aforementioned GMail situation.  That also started as a fun hack at Google as far as I know, then got turned into a Product TM.

So white knighting this project benefits the fediverse in exactly zero ways.

**Comment by raingloom** [comment]
The CF employees who started this should probably learn that to love something, you sometimes have to let it go. :)

**Comment by ocdtrekkie** [comment]
> The CF employees who started this should probably learn that to love something, you sometimes have to let it go. :)

They did, it's Apache licensed. Fork it and go nuts.

**Comment by ThatOneCalculator** [comment]
The code is wrapped around Cloudflare workers.

**Comment by Saiv46** [comment]
Ahem!

https://github.com/cloudflare/workerd

**Comment by NoaHimesaka1873** [comment]
> > I think we are very far from a point where a corporate implementation has standards-dictating authority.
> 
> 
> 
> And we should only get farther. :)
> 
> 
> 
> What the original devs intend matters little, *if* it gets enough steam, we have the aforementioned GMail situation.  That also started as a fun hack at Google as far as I know, then got turned into a Product TM.
> 
> 
> 
> So white knighting this project benefits the fediverse in exactly zero ways.

But gatekeeping fedi will certainly give bad impact. I don't understand fedi purists. It's like seeing 1930s Germany.

**Comment by NoaHimesaka1873** [comment]
Still, ignoring all the other problems, this is very unlikely to gain enough traction to perform the ol' classic EEE. Since it still requires to get domain and set it up yourself, not many people would be able to set up and use this. The most concerning thing is Tumblr supporting APub. Because:
1. Tumblr is easy to join vs Wildebeest requiring to buy domain and configure everything
2. Tumblr is free to join vs Wildebeest requires money
Personal note: It will be cool to see this ported to other serverless platforms, but this seems PITA since all platforms have different APIs and protocoals and whatnots.

**Comment by tootbrute** [comment]
> Because mastodon and the fediverse are federated, not centralized on a particular provider platform.
> 
> Discord killed the concept of "server" please dont kill federation.

this is a pointless issue. a more productive thing to do is to open an issue in Mastodon, Pleorama, Gotosocial and other fediverse implementations repo's and ask for an easy way to block (defederate) any instances that use this cloudflare 'activity pub' server.
